aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -503,47 +515,28 @@ public void query(
                     return;
                 }
 
-                LOG.debug("Deleting item in table: " + sqliteTable.getName() +
-                    " identified by ID: " + item.getId());
-
-                // delete always checks for ID first
-                final QueryPredicateOperation<?> idCheck =
-                    QueryField.field(primaryKeyName).eq(item.getId());
-                final QueryPredicate condition = !QueryPredicates.all().equals(predicate)
-                    ? idCheck.and(predicate)
-                    : idCheck;
-                final SqlCommand sqlCommand = sqlCommandFactory.deleteFor(modelSchema, condition);
-                if (sqlCommand.sqlStatement() == null || !sqlCommand.hasCompiledSqlStatement()) {
-                    onError.accept(new DataStoreException(
-                        "No delete statement found for the Model: " + modelSchema.getName(),
-                        AmplifyException.TODO_RECOVERY_SUGGESTION
-                    ));
-                    return;
-                }
-
-                synchronized (sqlCommand.getCompiledSqlStatement()) {
-                    final SQLiteStatement compiledSqlStatement = sqlCommand.getCompiledSqlStatement();
-                    compiledSqlStatement.clearBindings();
-                    bindStatementToValues(sqlCommand, null);
-                    // executeUpdateDelete returns the number of rows affected.
-                    final int rowsDeleted = compiledSqlStatement.executeUpdateDelete();
-                    compiledSqlStatement.clearBindings();
-                    if (rowsDeleted == 0) {
-                        throw new DataStoreException(
-                            "Failed to meet condition. Model was not deleted.",
-                            "Please verify the current state of saved item."
-                        );
+                // Use ModelTreeHelper to identify the models affected by cascading delete.
+                Map<ModelSchema, Set<String>> modelFamilyTree =
+                        sqLiteModelTree.descendantsOf(Collections.singleton(item));
+
+                for (ModelSchema schema : modelFamilyTree.keySet()) {
+                    for (String id : modelFamilyTree.get(schema)) {
+                        // Publish DELETE mutation for each affected item.
+                        String dummyJson = String.format("{\"id\":%s}", id);
```

Comment:
Quotes around %s?

Reply:
Actually, instead of manually writing JSON, create a `Map`, and then use gson to convert it to a `String`, like this:
```suggestion
                        String dummyJson = gson.toJson(Collections.singletonMap("id", id));
```

Reply:
clever! I prefer this approach more.